### DISCUSSION TITLE
Connector Type of Interfaces

### DISCUSSION THREAD

**Discussion by sw8899** [netbox-community/netbox#10364] [Q&A]
NetBox version 3.2.1

Is there a reason that it is possible to define a Connector Type (e.g. 8P8C, LC, MPO) on front and rear ports but not on interfaces?
Am I misunderstanding a concept or is it just missing on the interfaces?

**Comment by mtinberg** [comment]
I imagine it's usually less relevant for Interfaces, which already have a type that implies the connector in most cases, eg. a 10GBase-T interface is not going to be LC, but an SFP+ probably is (unless its DA ;-).  If you have a need to track something that isn't in the primary data model, you can custom fields to your heart's content to track whatever you need.  If enough people report something missing that maybe should be in the built-in models then it can be added and you can plan a migration to it.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: sw8899 ***@***.***>
Sent: Wednesday, September 14, 2022 10:49 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Connector Type of Interfaces (Discussion #10364)


NetBox version 3.2.1

Is there a reason that it is possible to define a Connector Type (e.g. 8P8C, LC, MPO) on front and rear ports but not on interfaces?
Am I misunderstanding a concept or is it just missing on the interfaces?

**Reply by sw8899** [reply]
I agree with 10GBase-T.
But I'm still not sure with for example a  QSFP+ (40GE). 
It can be LC or MTP (and MM or SM).
Custom fields are ok, but they don't show up in useful places like in the cable trace.

From my point of view this is very important information that is missing and should be addressed by a feature request.
But before I do that, I try to understand the logic that on a front port you can specify a Connector Type and not on a interface.

**Reply by DanSheps** [reply]
I don't thinik we want to get too far into the weeds on this one, otherwise we start modelling optics themselves which I am not sure we want to do.

**Reply by sw8899** [reply]
ok, maybe I got lost in translation. I'll try to repeat in my words what I think I understand:
In order to capture connector types on optical interfaces, the data model would have to be converted into a complex model, and that is something the maintainers of Netbox do not think is worth it.
Right?

**Reply by jcollie** [reply]
Two ways that kinda work around this:

1. Add a custom field to the interface that has the connector type. Not quite as good as a "real" field but should do the trick.
2. Add an inventory item that has the specific SFP, SFP+, or QSFP model number. This should implicitly specify the connector type as presumably the manufacturer of the SFP uses different model numbers for different connectors.

**Reply by sw8899** [reply]
thx for the input
1. ok
2. Since it is not possible to define inventory item **types** this would mean that you start over with the definition for every interface again?

**Reply by DanSheps** [reply]
I believe InventoryItemTypes were proposed at one point, not sure where that went.

But yes, you would need to define it for every interface.

**Comment by sw8899** [comment]
I understand that there are workarounds. But the main problem remains. They don't show up where you need them: cable trace, cables
My current problem is. The IT/Network departments do all the planning.
Then we (cable monkey department) go into netbox and see all the planned cables. 
Then we have to order and patch them.
I get the information Cat6a or OM4 and length and color and label from the cables list in Netbox.
But the most important property of the cable, the connector is missing.
From my point of view I can not understand how anybody can life without this information ;-)
![ConnectorType](https://user-images.githubusercontent.com/86421351/190453051-5270ee05-d559-4bfa-acfe-e90dacc2f3d6.JPG)

**Reply by DanSheps** [reply]
You should have your IT department do a custom export/view that gives you what you need.

**Reply by sw8899** [reply]
You mean, first implement the customs field and then export it with the connections/cables? 
As it is now the export would not make it better since the information isn't there.

Since the IT departments are spread around the globe, they don't care much about the cable monkeys view. So a regular (mandatory?) connector type field would give this topic the needed attention.

**Reply by shatt79** [reply]
Just have the networking team standardize on singlemode/LC across the board and you never need to worry or care about it.  There once was a cost benefit to using MM.  That doesn't exist anymore.  

There's literally nothing outside of GBICs or X2s that use anything but LC.  GBICS and X2s have been obsolete for a decade.

**Reply by sw8899** [reply]
I'm not in the position of giving the engineering teams advice.
I see hundreds of QSFP+ 40GBASE-SR4 **MTP** (e.g. in HP Synergy)

**Reply by jcollie** [reply]
> There once was a cost benefit to using MM. That doesn't exist anymore.

The cost of using existing multimode (where possible) will always be cheaper than installing new singlemode. There are those of us managing existing plants that are decades old. Plus there's all of the existing singlemode that was installed using SC or ST connectors. Telling people to spend millions on new fiber infrastructure to fix a software problem seems a bit out of touch.

**Reply by sw8899** [reply]
from FSdotCOM: QSFP-40G
MM MTP SR: $ 39.00
SM LC LR: $ 309.00 (cheapest SM) (Does SM SR exist?)

**Reply by candlerb** [reply]
> Does SM SR exist?

No: SR = Short Range = Multimode, pretty much by [definition](https://en.wikipedia.org/wiki/10_Gigabit_Ethernet#10GBASE-SR).

Multimode is inherently short range: the different propagation modes [interfere](https://en.wikipedia.org/wiki/Modal_dispersion) with each other.  Single-mode doesn't have this problem: its range is limited mainly by optical attenuation.

Both sides are right here.  Using existing multimode plant is clearly cheaper than deploying new single-mode, but all new deployments should be single-mode.  Even if the optics are a little more expensive, the single-mode fibre itself is cheaper and will accommodate any future bandwidth requirements.
